lib/net.js

```diff
@@ -1270,6 +1270,12 @@ function emitListeningNT(self) {
 }
 
 
+function throwIfInvalidPort(port) {
+  if (typeof port !== 'undefined' && !isLegalPort(port))
```

Comment:
If https://github.com/nodejs/node/issues/5733 lands, then the `typeof port !== 'undefined'` check would be unnecessary.